extensions/jackson/runtime/src/main/java/io/quarkus/jackson/runtime/ConfigurationCustomizer.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.jackson.runtime;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import io.quarkus.jackson.ObjectMapperCustomizer;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+
+import java.time.ZoneId;
+import java.util.TimeZone;
+
+@Singleton
+public class ConfigurationCustomizer implements ObjectMapperCustomizer {
+    @Inject
+    JacksonBuildTimeConfig jacksonBuildTimeConfig;
+
+    @Inject
+    JacksonSupport jacksonSupport;
+
+    @Override
+    public void customize(ObjectMapper objectMapper) {
+        if (!jacksonBuildTimeConfig.failOnUnknownProperties) {
+            // this feature is enabled by default, so we disable it
+            objectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        }
```

Comment:
Note that this code can be written a lot more elegantly as follows:
```suggestion
        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, jacksonBuildTimeConfig.failOnUnknownProperties);
```

But I wasn't sure whether this would create unwanted side-effects. Since these customizations have the highest priority, it should not make a difference - so I can easily convert them to the shorthand if you want.

Reply:
Let's keep it as it was for this PR